File: 2502-sort-the-people/sort-the-people.py
class Solution(object):
    def sortPeople(self, names, heights):
        """
        :type names: List[str]
        :type heights: List[int]
        :rtype: List[str]
        """
        d=dict(zip(heights,names))
        heights.sort(reverse=True)
        for i in range(len(heights)):
            names[i]=d[heights[i]]
        return names
        

File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& n, vector<int>& h) {
        unordered_map<int,string> u;
        if(n.size()==1){return n;}
        for(int i=0;i<h.size();i++){
            u[h[i]]=n[i];
        }
        sort(h.rbegin(),h.rend());
        for(int i=0;i<h.size();i++){
            n[i]=u[h[i]];
        }
        return n;
    }
};